pkg/internal/sqlsmith/setup.go

```diff
@@ -64,16 +65,20 @@ func randTables(r *rand.Rand) string {
 		SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
 		SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;
 		SET experimental_enable_enums = true;
+		SET experimental_partial_indexes = true;
 	`)
 
 	// Create the random tables.
 	stmts := sqlbase.RandCreateTables(r, "table", r.Intn(5)+1,
 		mutations.ForeignKeyMutator,
 		mutations.StatisticsMutator,
+		mutations.PartialIndexMutator,
 	)
 
+	fmtCtx := tree.NewFmtCtx(tree.FmtSerializable)
 	for _, stmt := range stmts {
-		sb.WriteString(stmt.String())
+		fmtCtx.FormatNode(stmt)
```

Comment:
Some of the interesting types weren't being serialized correctly. For example, it would produce something like this for infinity:

```sql
CREATE TABLE t (a INT, INDEX (a) WHERE a < Infinity
```

Reply:
I can add a comment here if that'd be helpful.

Reply:
Hmm, looks like it's still broken actually. Just have a partial index created with `(table2.col2_6 = +Inf)` and the tests failed.

Reply:
It looks like the `tree.Serialize` method does exactly this for you? `sb.WriteString(tree.Serialize(stmt))`